codesign/lib/src/utils.dart

```diff
@@ -5,10 +5,45 @@
 import 'dart:io';
 
 import 'package:args/args.dart';
+import 'package:file/file.dart';
 import 'package:process/process.dart';
 
 import 'log.dart';
 
+const String gsCloudBaseUrl = r'gs://flutter_infra_release';
+
+Future<void> upload({
```

Comment:
These should be moved into a dedicated gsutil class to ensure they have test coverage, and can be easily faked

Reply:
Sure, I grouped the four functions together and refactored them into a class called `engineArtifactTransfer`. It would take in gcloud base url as the parameter for the class. The fake instances of upload and download classes remove the process commands, and only return based on exit code.